### PR TITLE
uv: 0.4.30 -> 0.5.8

### PR DESCRIPTION
**Pull request by adisbladis**
## Things done

This cherry picks changes done to `uv` on staging.
Uv updates should never have started going to staging and now that decision keeps leaking in to the `uv` update workflow.
The motivation here is to stop using the staging workflow for uv and go straight to master.

The relatively large rebuild count of `uv` is caused by a [home-assistant test setup hack](https://github.com/NixOS/nixpkgs/pull/155060), and not a reason to keep targeting staging.

See related links for more discussion/complaints:
- https://github.com/NixOS/nixpkgs/pull/362808
- https://www.reddit.com/r/NixOS/comments/1hjnich/curiousityquestion_staging_not_being_merged_to/
- https://github.com/NixOS/nixpkgs/pull/357113#issuecomment-2492577597

Note: While 0.5.11 is the latest release 0.5.8 is the latest release that supports rustc <1.83.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by MisileLab — APPROVED** [review]

**Comment by MisileLab** [comment]
> The relatively large rebuild count of uv is caused by a https://github.com/NixOS/nixpkgs/pull/155060, and not a reason to keep targeting staging.

I targeted `staging` because 24.11 release was preparing when I submit pull request.

File: pkgs/by-name/uv/uv/package.nix [hunk 1]
```diff
@@ -47,29 +38,32 @@ python3Packages.buildPythonApplication rec {
     "uv"
   ];
 
-  postInstall = ''
+  # Tests require python3
+  doCheck = false;
```
Comment (lucasew) [inline pkgs/by-name/uv/uv/package.nix:42]: checkInputs?

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

- **Extra args**: ``
- **Free space**: true
- **Who asked**: lucasew

**TARGETS**
- **x86_64-linux**: true
- **x86_64-darwin**: true
- **aarch64-darwin**: true

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12454837299

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367280`

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
    <li>python312Packages.bentoml</li>
    <li>python312Packages.bentoml.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>hatch</li>
    <li>hatch.dist</li>
    <li>openllm</li>
    <li>openllm.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>python311Packages.uv</li>
    <li>python311Packages.uv.dist</li>
    <li>python312Packages.uv</li>
    <li>python312Packages.uv.dist</li>
    <li>uv</li>
  </ul>
</details>

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367280`

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
    <li>python312Packages.bentoml</li>
    <li>python312Packages.bentoml.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>hatch</li>
    <li>hatch.dist</li>
    <li>openllm</li>
    <li>openllm.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>python311Packages.uv</li>
    <li>python311Packages.uv.dist</li>
    <li>python312Packages.uv</li>
    <li>python312Packages.uv.dist</li>
    <li>uv</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
I don't think this PR is worth merging.
`staging-next` is expected to be merged either today or tomorrow.

**Pull request closed without merging**